### DISCUSSION TITLE
Issue with Reflecting Theme Data Changes in Keycloak Without Pod Restart

### DISCUSSION THREAD

**Discussion by nitindhiman314e** [keycloak/keycloak#35371] [Q&A]
### Description

We are using Keycloak in a Kubernetes (K8s) setup, where the Keycloak instance is managed by the Keycloak operator. We are also utilizing custom themes, which are stored in a persistent volume claim (PVC). However, we are encountering an issue where changes made to the theme data are not reflected in Keycloak until the pod is manually restarted. This behavior disrupts the continuous deployment process, as changes to themes require a pod restart to take effect.
We would like to explore solutions to apply rolling updates or a blue-green deployment strategy to ensure that theme data changes are automatically reflected without requiring a pod restart. Please suggest.

**Comment by ahus1** [comment]
The Keycloak operator will in version 26.2 support rolling updates for changed images where only the theme has changed. See https://github.com/keycloak/keycloak/issues/36305#issuecomment-2739430716 for details. 

In your setup, I would usually recommend a custom built Keycloak image which includes the theme instead of a PVC to utilize that functionality we added for the upcoming 26.2 release.
